custom_components/flashforge_adventurer_3/protocol.py

```diff
@@ -1,30 +1,38 @@
 import asyncio
 from codecs import StreamReader, StreamWriter
 import logging
-import os
 import re
 from typing import Optional, Tuple, TypedDict
 
 logger = logging.getLogger(__name__)
 
 BUFFER_SIZE = 1024
 TIMEOUT_SECONDS = 5
-STATUS_COMMAND = '~M601 S1'
-PRINT_JOB_INFO_COMMAND = '~M27'
-TEMPERATURE_COMMAND = '~M105'
+STATUS_COMMAND = '~M119\r\n'
+PRINT_JOB_INFO_COMMAND = '~M27\r\n'
+TEMPERATURE_COMMAND = '~M105\r\n'
+
+# Regular expressions for temperature and progress
+
+TEMPERATURE_REGEX_5M_PRO = re.compile(r'T0:(\d+\.?\d*)/(\d+\.?\d*) .* B:(\d+\.?\d*)/(\d+\.?\d*)')
+PROGRESS_REGEX_5M_PRO = re.compile(r'(\d+)/(\d+)\r')
```

Comment:
Since it's for both 5M and 5M Pro, let's use the `_5M` suffix (rather than `_5M_PRO`).